### PR TITLE
Clean up usages of config maps in pipelinerun_test

### PR DESCRIPTION
**Pull request by lbernick**
# Changes
This commit adds helper functions for creating config maps with various feature flags
and updates pipelinerun_test.go to use these helper functions.

/kind cleanup

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- n/a [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [x] [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Release notes block below has been filled in
(if there are no user facing changes, use release note "NONE")

# Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by lbernick** [comment]
assign @abayer 

Hoping this will help with #4734!

**Comment by abayer** [comment]
/lgtm

I forgot that I’m not in OWNERS. =)

File: pkg/reconciler/pipelinerun/pipelinerun_test.go [hunk 1]
```diff
@@ -901,14 +903,7 @@ func TestReconcile_CustomTask(t *testing.T) {
...
 
-	cms := []*corev1.ConfigMap{
-		{
-			ObjectMeta: metav1.ObjectMeta{Name: config.GetFeatureFlagsConfigName(), Namespace: system.Namespace()},
-			Data: map[string]string{
-				"enable-custom-tasks": "true",
-			},
-		},
-	}
+	cms := []*corev1.ConfigMap{withCustomTasks(newFeatureFlagsConfigMap())}
```
Comment (afrittoli) [inline pkg/reconciler/pipelinerun/pipelinerun_test.go:906]: Neat :)

**Review by afrittoli — COMMENTED** [review]
Thanks!
/lgtm

**Pull request merged**